api/v1beta1/openstackprovisionserver_webhook.go

```diff
@@ -115,4 +115,14 @@ func (r *OpenStackProvisionServer) Default() {
 	if r.Spec.OSImage == "" {
 		r.Spec.OSImage = openstackProvisionServerDefaults.OSImage
 	}
+	if r.Spec.Port == 0 {
```

Comment:
nit: we can remove some redundant/unreachable code[1] after this change.

[1] https://github.com/openstack-k8s-operators/openstack-baremetal-operator/blob/main/api/v1beta1/openstackprovisionserver.go#L43-L46

Reply:
Removed the redundant code